src/pages/signup/__tests__/signup.component.test.js

```diff
@@ -0,0 +1,89 @@
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import wrapper from 'test/test-utils';
+
+import SignUp from '../signup.component';
+
+let mockIsLoading = false;
+
+jest.mock('service/signup', () => ({
+  useMutationSignup: () => ({
+    isLoading: mockIsLoading,
+  }),
+}));
+
+describe('<SignUp />', () => {
+  it('renders the title and all the inputs', () => {
+    wrapper(<SignUp />);
+    expect(
+      screen.getByRole('heading', { name: /cadastre-se/i })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /nome completo/i })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /e-mail/i })
+    ).toBeInTheDocument();
+    expect(screen.getByRole('textbox', { name: /cpf/i })).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /telefone/i })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /endereço/i })
+    ).toBeInTheDocument();
+    expect(screen.getByLabelText(/^senha/i)).toBeInTheDocument();
+    expect(screen.getByLabelText(/confirmar senha/i)).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: /cadastrar/i })
+    ).toBeInTheDocument();
+  });
+
+  it('first renders with a disabled submit button', () => {
+    wrapper(<SignUp />);
+    expect(screen.getByRole('button', { name: /cadastrar/i })).toBeDisabled();
+  });
+
+  it('should enable submit button when validation passes', async () => {
+    wrapper(<SignUp />);
+    const full_name = screen.getByRole('textbox', { name: /nome completo/i });
```

Comment:
In JS we barely use snake case:
```suggestion
    const fullName = screen.getByRole('textbox', { name: /nome completo/i });
```